bindings/python/pyiree/compiler2/xla.py

```diff
@@ -0,0 +1,184 @@
+# Lint-as: python3
+"""XLA compiler interface."""
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(#4131) python>=3.7: Use postponed type annotations.
+
+from enum import Enum
+import logging
+import tempfile
+from typing import List, Optional, Sequence, Set, Union
+
+from .tools import find_tool, invoke_immediate, invoke_pipeline
+from .core import CompilerOptions, DEFAULT_TESTING_BACKENDS, build_compile_command_line
+
+__all__ = [
+    "compile_file",
+    "compile_str",
+    "is_available",
+    "DEFAULT_TESTING_BACKENDS",
+    "ImportOptions",
+    "ImportFormat",
+]
+
+_IMPORT_TOOL = "iree-import-xla"
+
+
+def is_available():
+  """Determine if the XLA frontend is available."""
+  try:
+    find_tool(_IMPORT_TOOL)
+  except ValueError:
+    logging.warning("Unable to find IREE tool %s", _IMPORT_TOOL)
+    return False
+  return True
+
+
+class ImportFormat(Enum):
+  """Import type of the model."""
+  BINARY_PROTO = "binary_proto"
```

Comment:
Do you see other variants here beyond proto-in-some-form? An ergonomics improvement (in a follow-up etc) would be to remove this and instead just snoop the first few bytes to see if it's text or binary. IIRC the text proto forms always start with "hlo_module {".

Reply:
(this would go in the cpp file -- and I think I have some old code internally that did this efficiently we could steal)

Reply:
There was an HLO Text format that I was looking at but backed out of. I'll defer on magical proto parsing for this patch (also: not necessarily yours, but I have trauma from debugging too many hand-rolled proto parsing switches like that... such a bad API).